api-bucket-replication.go

```diff
@@ -289,3 +289,53 @@ func (c *Client) GetBucketReplicationResyncStatus(ctx context.Context, bucketNam
 	}
 	return rinfo, nil
 }
+
+// BucketReplicationDiff - gets diff for unreplicated entries.
+func (c *Client) BucketReplicationDiff(ctx context.Context, bucketName string, opts replication.DiffOpts) <-chan replication.DiffInfo {
+	diffCh := make(chan replication.DiffInfo)
+
+	// start a routine to start reading line by line.
+	go func(diffCh chan<- replication.DiffInfo) {
+		defer close(diffCh)
+		urlValues := make(url.Values)
+		urlValues.Set("replication-diff", "")
+
+		if opts.Verbose {
+			urlValues.Set("verbose", "true")
+		}
+		urlValues.Set("arn", opts.ARN)
+		urlValues.Set("prefix", opts.Prefix)
+		// Execute GET on bucket to get replication config.
+		resp, err := c.executeMethod(ctx, http.MethodPost, requestMetadata{
+			bucketName:  bucketName,
+			queryValues: urlValues,
+		})
+		if err != nil {
+			diffCh <- replication.DiffInfo{Err: err}
+			return
+		}
+
+		if resp.StatusCode != http.StatusOK {
+			closeResponse(resp)
```

Comment:
you can do `defer closeResponse(resp)` instead of calling it everywhere, since we shall close response body only upon return.

Reply:
done